docs/source/models/field_ionization_charge_state_prediction.py

```diff
@@ -0,0 +1,315 @@
+                time / fs, rate_matrix[i, :],
+                label="{}+ to {}+".format(
+                    Z[i] - 1, Z[i]), color=color[i]
+            )
+            ax_bsi.axvline(
+                time_BSI[i] / fs,
+                label="{}+ to {}+".format(
+                    Z[i] - 1, Z[i]), color=color[i]
+            )
+
+        ax_pop.plot(
+            time / fs, charge_dist[i, :-1] / percent,
+            label="{}+".format(i), color=color[i]
+        )
+        ax_pop.fill_between(
+            x=time / fs,
+            y1=0, y2=charge_dist[i, :-1] / percent,
+            color=color[i], alpha=.3
+        )
+        # color the regions between charge state transitions in the BSI model
+        if (i > 0 and i < Z_max):
+            ax_bsi.fill_betweenx(
+                y=yfill_range,
+                x1=time_BSI[i - 1] / fs,
+                x2=time_BSI[i] / fs,
+                color=color[i],
+                alpha=.3
+            )
+        # color the range between the earliest time and the first charge state
+        # transition
+        if (i == 0):
+            ax_bsi.fill_betweenx(
+                y=yfill_range,
+                x1=time[0] / fs,
+                x2=time_BSI[i] / fs,
+                color=color[i],
+                alpha=.3
+            )
+        # color the range between the last charge state transition and the
+        # latest time
+        if (i == Z_max):
+            ax_bsi.fill_betweenx(
+                y=yfill_range,
+                x1=time_BSI[i - 1] / fs,
+                x2=time[-1] / fs,
+                color=color[i],
+                alpha=.3
+            )
+
+    ax_bsi_twin = ax_bsi.twinx()
+    # plot intensity in W/cm^2
+    ax_bsi_twin.plot(time / fs, intensity_envelope_SI / 1e4, c="k")
+
+    locs = np.array([0, 100])
+    labels = ["none", "all"]
+    ax_bsi.set_yticks(locs)
+    ax_bsi.set_yticklabels(labels)
+
+    # secondy y-axis for rate plot
+    secaxy_rate = ax_rate.secondary_yaxis(
+        'right',
+        functions=(time_SI_to_AU, time_AU_to_SI)
+    )
+    secaxy_rate.set_ylabel(r'ionizations per second')
+
+    # set plot limits
+    ax_rate.set_xlim(xlim)
+    ax_rate.set_yscale("log")
+    ax_rate.set_ylim(ylim_ax_rate)
+    ax_pop.set_ylim(ylim_ax_pop)
+    ax_bsi.set_ylim(ylim_ax_bsi)
+    ax_bsi_twin.set_yscale("log")
+    ax_bsi_twin.set_ylim(ylim_ax_bsi_twin)
+
+    # note string for ADK rate plot note
+    note_string = "Note: ADK rates were calculated from the " \
+        + "intensity envelope below"
+    # note in ADK rate plot
+    ax_rate.text(
+        x=0.05,
+        y=0.9,
+        s=note_string,
+        fontsize=12,
+        transform=ax_rate.transAxes
+    )
+
+    # labels
+    ax_rate.set_ylabel(r"ADK: Ionization Rate $\Gamma\,\mathrm{(AU^{-1})}$")
+    ax_pop.set_ylabel(r"ADK: Relative Population (%)")
+    ax_bsi_twin.set_ylabel(r"Laser Intensity $I\,\mathrm{(W/cm^2)}$")
+    ax_bsi.set_xlabel(r"Time $t-t_\mathrm{max}\,\mathrm{(fs)}$")
```

Comment:
The ticks numbers without labels confuse a bit the fact, that you have a shared axis. Just remove the ticks numbers for the upper tow plots x-axis. 

```suggestion
    ax_rate.xticks([])
    ax_pop.xticks([])
    ax_bsi.set_xlabel(r"Time $t-t_\mathrm{max}\,\mathrm{(fs)}$")
```

Reply:
@n01r If you like the numbers in between - I am fine with that as well. 
(my suggestions removes not only the ticks labels, but the ticks as well - this is definitely not good. There should be better option by just removing the tick labels and keeping the ticks.)

Reply:
*offline discussion with @n01r:* He preserve the numbers in between.